ts/sagas/startup/generateCryptoKeyPair.ts

```diff
@@ -0,0 +1,32 @@
+import { generate, deleteKey, CryptoError } from "io-react-native-crypto";
+import { call } from "typed-redux-saga/macro";
+import {
+  checkPublicKeyExists,
+  isKeyAlreadyGenerated,
+  setKeyAlreadyGenerated
+} from "../../utils/crypto";
+import { mixpanelTrack } from "./../../mixpanel";
+
+const KEY_NAME = "lp-temp-key";
+
+export function* generateCryptoKeyPair() {
+  try {
+    const keyExistsOnKeystore = yield* call(checkPublicKeyExists, KEY_NAME);
+    const keyAlreadyExists = yield* call(isKeyAlreadyGenerated, KEY_NAME);
+    const generateKey = !(keyAlreadyExists || keyExistsOnKeystore);
+
+    if (generateKey) {
+      const key = yield* call(generate, KEY_NAME);
+      void mixpanelTrack("LOLLIPOP_KEY_GENERATION_SUCCESS", {
+        kty: key.kty
+      });
+      yield* call(setKeyAlreadyGenerated, KEY_NAME, key.kty);
+      yield* call(deleteKey, KEY_NAME);
```

Comment:
What happen here with deleteKey?

Reply:
The key is removed from the keystore after it's been generated successfully. This is only a test. We leave nothing on the device after this test.